XiEditor/XiCore.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+protocol XiCore {
```

Comment:
I would add a short `/// doc comment` here (and on other new classes) and I would also add docs for the members of this protocol, since it's a pretty important part of the system and people will probably want to look it up regularly.

Reply:
done 👍